pandas/core/reshape/pivot.py

```diff
@@ -534,31 +537,41 @@ def crosstab(
     b      0  1  0
     c      0  0  0
     """
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
+
+    if values is not None and aggfunc is None:
+        raise ValueError("values cannot be used without an aggfunc.")
 
     index = com.maybe_make_list(index)
     columns = com.maybe_make_list(columns)
 
-    rownames = _get_names(index, rownames, prefix="row")
-    colnames = _get_names(columns, colnames, prefix="col")
-
     common_idx = None
     pass_objs = [x for x in index + columns if isinstance(x, (ABCSeries, ABCDataFrame))]
     if pass_objs:
         common_idx = get_objs_combined_axis(pass_objs, intersect=True, sort=False)
 
-    data = {}
-    data.update(zip(rownames, index))
-    data.update(zip(colnames, columns))
-
-    if values is None and aggfunc is not None:
-        raise ValueError("aggfunc cannot be used without values.")
+    rownames = _get_names(index, rownames, prefix="row")
+    colnames = _get_names(columns, colnames, prefix="col")
 
-    if values is not None and aggfunc is None:
-        raise ValueError("values cannot be used without an aggfunc.")
+    # We create our own mapping of row and columns names
+    # to prevent issues with duplicate columns. GH Issue: #22529
+    shared_col_row_names = set(rownames).intersection(set(colnames))
```

Comment:
rather than intersect I would raise if duplicates are passed

Reply:
There are three sources of duplicates: 
- Duplicate column names
- Duplicate row names
- Names shared between rows and columns

My initial proposal raised an error in all cases, the current proposal supports all of them. You want it two work for duplicate column/row names but not for names shared between them?